namespace/namespace.go

```diff
@@ -39,6 +44,27 @@ func MustNew(version uint8, id []byte) Namespace {
 	return ns
 }
 
+// NewFromBytes returns a new namespace from the provided byte slice.
+func NewFromBytes(bytes []byte) (Namespace, error) {
+	if len(bytes) != NamespaceSize {
+		return Namespace{}, fmt.Errorf("invalid namespace length: %v must be %v", len(bytes), NamespaceSize)
+	}
+
+	err := validateVersionSupported(bytes[0])
+	if err != nil {
+		return Namespace{}, err
```

Comment:
Should we wrap error with additional info what was wrong? Something like:

```
return Namespace{}, fmt.Errorf("invalid version: %v", bytes[0])
```

Same for the `validateID` below.

Reply:
The errors seem descriptive enough themselves